libraries/rush-lib/src/logic/pnpm/PnpmLockfileConfiguration.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { AlreadyReportedError, Encoding } from '@rushstack/node-core-library';
+import { FileSystem } from '@rushstack/node-core-library';
+import yaml from 'js-yaml';
+import type { Lockfile } from '@pnpm/lockfile-types';
+import type { ITerminal } from '@rushstack/terminal';
+
+import type { RushConfiguration } from '../../api/RushConfiguration';
+import { CustomTipId } from '../../api/CustomTipsConfiguration';
+import type { PnpmLockfilePolicy } from './PnpmOptionsConfiguration';
+
+export class PnpmLockfileConfiguration {
+  /**
+   * Determine whether `pnpm-lock.yaml` complies with the rules specified in `common/config/rush/pnpm-config.schema.json`.
+   * @internal
+   */
+  public static validateLockfile(
+    terminal: ITerminal,
+    rushConfiguration: RushConfiguration,
+    filename: string
+  ): void {
+    const pnpmLockfilePolicies: PnpmLockfilePolicy[] | undefined =
+      rushConfiguration.pnpmOptions.pnpmLockfilePolicies;
+
+    if (pnpmLockfilePolicies && pnpmLockfilePolicies.length > 0) {
+      const lockfileRawContent: string = FileSystem.readFile(filename, { encoding: Encoding.Utf8 });
+      const lockfile: Lockfile = yaml.load(lockfileRawContent);
+
+      for (const policy of pnpmLockfilePolicies) {
+        PnpmLockfileConfiguration[policy](terminal, rushConfiguration, lockfile);
+      }
+    }
+  }
+
+  /**
+   * Determine whether `pnpm-lock.yaml` contains insecure sha1.
+   * @internal
+   */
+  public static disallowInsecureSha1(
+    terminal: ITerminal,
+    rushConfiguration: RushConfiguration,
+    lockfile: Lockfile
+  ): void {
+    const { packages } = lockfile;
+    if (packages) {
+      for (const { resolution } of Object.values(packages)) {
+        if ((resolution as { integrity: string }).integrity.startsWith('sha1')) {
+          rushConfiguration.customTipsConfiguration._showErrorTip(
```

Comment:
What happens if someone turns this option on but doesn't have a custom tip for it configured?

Seems like this should just be a standard error message.

Reply:
![image](https://github.com/microsoft/rushstack/assets/68707557/d8cb3078-f6b9-451c-b7d5-5fd631ecc58f)
No output. Do you think we should give a default message when user doesn't config a custuom tip?

Reply:
Yeah. If someone turns this option on, there should be something printed if the rule is violated, even if they don't have their own message configured.

Reply:
Here's an example of how `TIP_RUSH_INCONSISTENT_VERSIONS ` does that:

https://github.com/microsoft/rushstack/blob/caab398bce0a67e71aac48af9a7642955c387abf/libraries/rush-lib/src/logic/versionMismatch/VersionMismatchFinder.ts#L147-L159

Reply:
> *intended to provide additional detail for existing error messages*

Sorry this was unclear. In 97f17e7c6b08094e2780e6fdd672f08898d59f40 I've better documented that Rush's standard message is not replaced by custom tips.

The UX now looks like this:

![image](https://github.com/microsoft/rushstack/assets/4673363/9ea38be6-f20e-4631-8e3d-bc048f7fb565)